packages/core/src/components/overflow-list/overflowList.tsx

```diff
@@ -118,14 +141,28 @@ export class OverflowList<T> extends React.PureComponent<IOverflowListProps<T>,
         ) {
             // reset visible state if the above props change.
             this.setState({
+                direction: OverflowDirection.GROW,
+                lastOverflow: undefined,
                 overflow: [],
                 visible: nextProps.items,
             });
         }
     }
 
-    public componentDidUpdate() {
+    public componentDidUpdate(_prevProps: IOverflowListProps<T>, prevState: IOverflowListState<T>) {
         this.repartition(false);
+        if (
+            // if a resize operation has just completed (transition to NONE)
+            this.state.direction === OverflowDirection.NONE &&
+            this.state.direction !== prevState.direction &&
+            // shrink operation does not require checking last items
+            (prevState.direction === OverflowDirection.SHRINK ||
+                // if current overflow is different from last completed state
+                this.state.lastOverflow == null ||
+                this.state.lastOverflow.some((x, i) => x !== this.state.overflow[i]))
```

Comment:
Could we just check `lastOverflow.length !== overflow.length` to save a little time? Is it possible for a change in overflow to replace items without changing the number of items displayed?

Reply:
yes it's completely possible for that to happen, though that should be handled by the check in `componentWillReceiveProps`. so maybe this check can be made simpler. i'll play with it some more.

Reply:
worked flawlessly!